referenceframe/frame_system.go

```diff
@@ -317,58 +316,37 @@ func (sfs *simpleFrameSystem) FrameSystemSubset(newRoot Frame) (FrameSystem, err
 // the point representation of their geometry type with respect to the world.
 func FrameSystemToPCD(system FrameSystem, inputs map[string][]Input, logger golog.Logger) (map[string][]r3.Vector, error) {
 	vectorMap := make(map[string][]r3.Vector)
-	geoMap, err := FrameSystemGeometries(system, inputs, logger)
+	geometries, err := FrameSystemGeometries(system, inputs)
 	if err != nil {
-		return nil, err
+		logger.Debug(err)
 	}
-	for name, geosInFrame := range geoMap {
-		geos := geosInFrame.Geometries()
-		aggregatePoints := []r3.Vector{}
-		for _, g := range geos {
-			asPoints := g.ToPoints(defaultPointDensity)
-			aggregatePoints = append(aggregatePoints, asPoints...)
-		}
-		vectorMap[name] = aggregatePoints
+	for _, geometry := range geometries.Geometries() {
+		vectorMap[geometry.Label()] = geometry.ToPoints(defaultPointDensity)
 	}
 	return vectorMap, nil
 }
 
 // FrameSystemGeometries takes in a framesystem and returns a map where all elements
 // are GeometriesInFrame modified to be with respect to the world.
-func FrameSystemGeometries(system FrameSystem, inputs map[string][]Input, logger golog.Logger) (map[string]*GeometriesInFrame, error) {
-	geoMap := make(map[string]*GeometriesInFrame)
+func FrameSystemGeometries(system FrameSystem, inputs map[string][]Input) (*GeometriesInFrame, error) {
+	var errAll error
+	allGeometries := make([]spatial.Geometry, 0)
 	for _, name := range system.FrameNames() {
-		currentFrame := system.Frame(name)
-		currentInput := inputs[name]
-		// if currentInput is nil and DoF != 0 we chose to omit the
-		// frame entirely as this would return the frame's geometries
-		// in their home or "zero" position, and not in their
-		// current position.
-		if currentInput == nil && len(currentFrame.DoF()) == 0 {
-			currentInput = []Input{}
-		}
-		if currentInput == nil {
-			logger.Debugf("will not transform %v to be with respect to the world as it had no inputs provided", name)
-			continue
-		}
-		geosInFrame, err := currentFrame.Geometries(currentInput)
+		geosInFrame, err := system.Frame(name).Geometries(inputs[name])
 		if err != nil {
-			return nil, err
+			errAll = multierr.Append(errAll, err)	
```

Comment:
same here - why not just return the err.

Reply:
In this case, I think it's a good idea. We should be letting the caller, and ultimately the user, know about every problem we encounter, not just the first. We might have a `Geometries` error on one frame and a `Transform` error on a different frame, and it would be good to address those both at once.

Reply:
ok I see what you are saying.
My thinking was that if we have frame `a` and frame `b`, such that `b`'s parent is `a`, then a transform error for `a` will return in error for `b`.

Reply:
The original logic was logging every issue as a separate line, this is the equivalent that treats them as errors returned to the calling function, which can then log them